src/libraries/System.Private.CoreLib/src/System/Threading/WaitHandle.Windows.cs

```diff
@@ -49,20 +49,49 @@ private static unsafe int WaitForMultipleObjectsIgnoringSyncContext(IntPtr* pHan
             Thread currentThread = Thread.CurrentThread;
             currentThread.SetWaitSleepJoinState();
 
+            ulong startTime = 0;
+            if (millisecondsTimeout != -1)
+            {
+                startTime = Interop.Kernel32.GetTickCount64();
+            }
+
+        retry:
 #if NATIVEAOT
             int result;
             if (reentrantWait)
             {
                 Debug.Assert(!waitAll);
-                result = RuntimeImports.RhCompatibleReentrantWaitAny(false, millisecondsTimeout, numHandles, pHandles);
+                result = RuntimeImports.RhCompatibleReentrantWaitAny(true, millisecondsTimeout, numHandles, pHandles);
             }
             else
             {
-                result = (int)Interop.Kernel32.WaitForMultipleObjectsEx((uint)numHandles, (IntPtr)pHandles, waitAll ? Interop.BOOL.TRUE : Interop.BOOL.FALSE, (uint)millisecondsTimeout, Interop.BOOL.FALSE);
+                result = (int)Interop.Kernel32.WaitForMultipleObjectsEx((uint)numHandles, (IntPtr)pHandles, waitAll ? Interop.BOOL.TRUE : Interop.BOOL.FALSE, (uint)millisecondsTimeout, Interop.BOOL.TRUE);
             }
 #else
-            int result = (int)Interop.Kernel32.WaitForMultipleObjectsEx((uint)numHandles, (IntPtr)pHandles, waitAll ? Interop.BOOL.TRUE : Interop.BOOL.FALSE, (uint)millisecondsTimeout, Interop.BOOL.FALSE);
+            int result = (int)Interop.Kernel32.WaitForMultipleObjectsEx((uint)numHandles, (IntPtr)pHandles, waitAll ? Interop.BOOL.TRUE : Interop.BOOL.FALSE, (uint)millisecondsTimeout, Interop.BOOL.TRUE);
 #endif
+
+            if (result == Interop.Kernel32.WAIT_IO_COMPLETION)
+            {
+                // Handle APC completion by adjusting timeout and retrying
+                if (millisecondsTimeout != -1)
+                {
+                    ulong currentTime = Interop.Kernel32.GetTickCount64();
+                    ulong elapsed = currentTime - startTime;
+                    
```

Comment:
Some CI build legs are failing due to trailing whitespaces. Make sure there are no trailing whitespace on any line in the delta.